// @whcg/whcg-chart/dist/index.es6.js
import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';

/**
 * `WhcgChart`
 * 
 * @customElement
 * @polymer
 */

class WhcgChart extends PolymerElement {
    static get template() {
        return html `
        <style>
        :host {
            display: inline-block;
        }
        </style>
    <div>
        <!--<canvas id="myChart" width$="{{width}}" height$="{{height}}"></canvas>-->
        <canvas id="myChart" style="width: 100%; height: 100%"></canvas>
    </div>
        `
    }
    static get properties() {
        return {
            type: {
                type: String,
                notify: false,
                readOnly: false,
            },
    
            chartjson: {
                type: String,
                notify: false,
                readOnly: false,
                observer: '_chartjsonChanged'
            },
            width: {
                type: String,
                notify: false,
                readOnly: false,
                value: '400px' 
            },
            height: {
                type: String,
                notify: false,
                readOnly: false,
                value: '400px' 
            },
            legendposition: {
                type: String,
                notify: false,
                readOnly: false,  
                value: 'right'
            },

            legendfontsize: {
                type: Number,
                notify: false,
                readOnly: false,
                value: 12  
            },

            legendfontfamily: {
                type: String,
                notify: false,
                readOnly: false,
                value: 'Arial' 
            },

            stacked: {
                type: Boolean,
                notify: false,
                readOnly: false,
                value: false 
            },
        }
    }

    _chartjsonChanged() {
        console.log('chartJsonChanged');
        this._chartJs(JSON.parse(this.chartjson));
    }

    _chartJs(data) {
        console.log('data');
        console.log(data);
        var ctx = this.$.myChart;
        ctx.style.backgroundColor = 'hsla(214, 50%, 22%, 0.1)';
        //Chart.defaults.global.defaultFontColor = 'white';
        //Chart.defaults.global.elements.rectangle.backgroundColor = 'hsla(24, 70%, 50%, 1)';
        //Chart.defaults.global.elements.rectangle.borderColor = '#FFFFFF';
        //Chart.defaults.global.elements.rectangle.borderWidth = 1;

        var myChart = new Chart(ctx, {
            type: this.type,
            data: data,

            options: {
                legend: {
                    position: this.legendposition,
                    labels: {
                        fontFamily: "'Exo 2', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
                        fontColor: '#FFFFFF',
                        fontSize: 14,
                        boxWidth: 14
                    }
                },
                scales: {
                    yAxes: [{
                        ticks: {
                            beginAtZero:true
                        }
                    }],
                    yAxes: [{
                        ticks: {
                            fontFamily: "'Exo 2', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
                            fontColor: '#FFFFFF',
                            fontSize: 14
                        },
                        gridLines: {
                            // display: true,
                            // color: '#FFFFFF'
                        }
                    }],
                    xAxes: [{
                        ticks: {
                            fontFamily: "'Exo 2', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
                            fontColor: '#FFFFFF',
                            fontSize: 14
                        },
                        gridLines: {
                            // display: true,
                            // color: '#FFFFFF'
                        }
                        // scaleLabel: { fontColor: '#6E6E6E', fontSize:16 }
                    }]
                },
                responsive: true,
                maintainAspectRatio: false,
                title: {
                    display: true,
                    text: 'Title',
                    fontFamily: "'Exo 2', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
                    fontColor: '#FFFFFF',
                    fontSize: 14
                }
                //  label: { fontSize:18 }
                
                // scales: {
                //     xAxes: [{
                //         stacked: this.stacked
                //     }],
                //     yAxes: [{
                //         stacked: this.stacked
                //     }
                //     // {
                //     //     ticks: {
                //     //         beginAtZero: true
                //     //     }
                //     // }
                // ]
                // }
            }
        });

    }
}

window.customElements.define('whcg-chart', WhcgChart);

export { WhcgChart };
