// @whcg/whcg-app/dist/index.es6.js
import '@polymer/polymer/lib/elements/custom-style.js';
import { PolymerElement, html } from "@polymer/polymer/polymer-element.js";
import "@polymer/polymer/lib/elements/dom-bind.js";
import "@polymer/polymer/lib/elements/dom-if.js";
import '@whcg/whcg-general-styles/grid.js';
import "@whcg/whcg-box-container";
import "@whcg/whcg-chart";
import "@whcg/whcg-chart-feeder";
import "@whcg/whcg-grid";
import "@whcg/whcg-lit-grid";
import "@whcg/whcg-button";
import "@whcg/whcg-json-merger";
import "@whcg/whcg-login-check";
import "@whcg/whcg-login-fields";
import "@whcg/whcg-select";
import "@whcg/whcg-footer";
import "@vaadin/vaadin-grid";
import "@vaadin/vaadin-grid/vaadin-grid-selection-column.js";
import "@vaadin/vaadin-grid/vaadin-grid-column.js";
import "@vaadin/vaadin-grid/vaadin-grid-sorter.js";
import "@vaadin/vaadin-grid/vaadin-grid-filter.js";
import "@whcg/whcg-period-summarizer";
import "@whcg/whcg-period-discounter";
import "@whcg/whcg-number-field-box";
import "@whcg/whcg-number-field";
import "@whcg/whcg-period-compounder";
import "@whcg/whcg-period-amplifier";
import "@whcg/whcg-multiplier";
import "@whcg/whcg-json-maker";
import "@whcg/whcg-section-text-input";
import "@whcg/whcg-section-input-text";
import "@whcg/whcg-section-textlong-input-chart";
import "@whcg/whcg-section-textlong-chart-input";
import "@whcg/whcg-section-chart-text-inputlong";
import "@whcg/whcg-section-text-chart-inputlong";
import "@whcg/whcg-section-textlong-chartlong";
import "@whcg/whcg-section-wide-chartlong";

class WhcgApp extends PolymerElement {

    constructor() {
        super();
        this.db = new PouchDB('http://localhost:5984/statistics');
        this.maintableitems = [];
    }
  static get properties() {
    return {
        maintableitems: {
            type: Array,
            readOnly: false,
            notify: true,
            observer: '_maintableitemsChanged'
        },
        brokertableitems: {
            type: Array,
            readOnly: false,
            notify: true,
            observer: '_brokertableitemsChanged'
        },

        municipalitytableitems: {
            type: Array,
            readOnly: false,
            notify: true,
            observer: '_municipalitytableitemsChanged'
        },
        selectedpage: {
            type: String,
            notify: true,
            readOnly: false,
            observer: '_valueChangedinApp'
        },
        period: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsSumAreasJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsSumDeclineAreasJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsSumExpansionAreasJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsNonCompoundedAreaAmountsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedHeatCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsDiscountedCompoundedHeatCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedRentCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsDiscountedCompoundedRentCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedDeclineCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsDiscountedCompoundedDeclineCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedExpandCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsDiscountedCompoundedExpandCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsTotalMergedCompoundedCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsTotalMergedDiscountedCompoundedCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsMergedDiscountedCompoundedCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsMergedCompoundedCostsJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedHeatCostsOwnJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsInitialEstablishCostOwn: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedRepairCostsPerSqmOwnJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsCompoundedRepairCostsOwnJson: {
            type: String,
            notify: true,
            readOnly: false,
        },
        chartJsSumMaintenanceAllJson: {
            type: String,
            notify: true,
            readOnly: false,
        }

        

        
    };
  }

  _isEqualTo(title, string) {
        return title == string;
    }

    // _chartJsSumExpansionAreasJsonChanged() {
    //     console.log('JSONCHANGED!');
    // }

  _maintableitemsChanged() {
      console.log("_maintableitemsChanged");
      console.log(this.maintableitems);
  }

    _brokertableitemsChanged() {
        console.log("_brokertableitemsChanged");
        console.log(this.brokertableitems);
    }


    _municipalitytableitemsChanged() {
        console.log("_municipalitytableitemsChanged");
        console.log(this.municipalitytableitems);
    }



  static get template() {
    return html`

	<style include = "style-element-grid">

.hide {
  position: absolute;

  width: 1px;
  height: 1px;
  margin: -1px;
  border: 0;
  padding: 0;

  white-space: nowrap;

  clip-path: inset(100%);
  clip: rect(0 0 0 0);
  overflow: hidden;
}

        .wrapper {
            display: grid;
            grid-template-columns: 1fr 1440px 1fr;
            background: linear-gradient(to bottom, #005F9A 100px,#005F9A 0vh,#63D1F4 300vh);
            background-attachment: fixed;
            min-height: 100vh;
        }

        .left {
            grid-column: 1;
        }

        .container {
            grid-column: 2;
        }

        .right {
            grid-column: 3;
        }

        @media only screen and (max-width: 1440px) {
            .wrapper {
            display: grid;
            grid-template-columns: 0px 1fr 0px;
            }

            .left {
                grid-column: 1;
            }

            .container {
                grid-column: 2;
            }

            .right {
                grid-column: 3;
            }
        }

        .header {
            padding-top: 20px;
            padding-bottom: 0px;
        }

        .main {
            padding-top: 200px;
            padding-bottom: 0px;
        }





        .mainrightpaddingtop {
            padding-top: 240px;
        }

        .headline {
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-xxxl);
            font-weight: var(--parmaco-font-weight-normal);
            color: var(--parmaco-base-color-100pct);
            padding-bottom: 10px;
        }

        .headline-big {
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-xxxxl);
            font-weight: var(--parmaco-font-weight-normal);
            color: var(--parmaco-base-color-100pct);
            padding-bottom: 60px;
            padding-top: 20px;
        }

        .ingress {
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-m);
            font-weight: var(--parmaco-font-weight-normal);
            color: var(--parmaco-base-color-100pct);
        }




        .categoryinputsection {
            padding-top: 130px;
        }

        .categoryinputsection-headline {
            padding-top: 32px;
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-xl);
            color: var(--parmaco-base-color-100pct);
        }
        .categoryinputsection-content {
        }

        .categoryinputsection-content--withchartbottom {
            grid-template-rows: auto 350px;
        }

        .categoryinputsection-content--withcharttop {
            grid-template-rows: 350px auto;
        }

        .categoryinputsection-content-text{
            padding-top: 33px;
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-s);
            font-weight: var(--parmaco-font-weight-normal);
            color: var(--parmaco-base-color-100pct);
        }

        .categoryinputsection-content-text--top{
            padding-top: 33px;
            font-family: var(--parmaco-font-family);
            font-size: var(--parmaco-font-size-s);
            font-weight: var(--parmaco-font-weight-normal);
            color: var(--parmaco-base-color-100pct);
        }

        .categoryinputsection-content-inputbox {
        }

        .categoryinputsection-content-inputboxrow--bottom {
            padding-top: 33px;
        }

        .categoryinputsection-content-chart {
            padding-top: 33px;
        }

        .categoryinputsection-content-chart--bottom {
            padding-top: 33px;
        }

        .categoryinputsection-content-chart--top {
            padding-top: 33px;
        }




        .gridcontent-right {
            justify-self: end;
        }


        .rent-expansion-wrapper {
            padding-top: 150px;
        }
        
        .rent-detraction-wrapper {
            padding-top: 150px;
        }

        .chart-box-wrapper {
            padding-top: 150px;
        }
        

        .footer {
            padding-top: 450px;
            padding-bottom: 0px;
        }
        
  </style>


    
	
    <div class="wrapper">
        <div class="left"></div>
        <div class="grid-12 container">
            <whcg-header class="col1span12 header"></whcg-header>
            
            <div class="col1span12 grid-12 main">
                 <div class="col4span8 headline-big">
                        KALKYLMODELL FÖR SKOLLOKALER
                    </div>
                <div class="col2span2 grid-2 sidemenu">
                    <whcg-lit-tab page="{{selectedpage}}"></whcg-lit-tab>
                </div>
                
                <div class="col4span8 grid-8 headlineandingress">
                        <iron-pages selected="{{selectedpage}}" attr-for-selected="key" fallback-selection="/" class="col1span8">
                            <div key="0">
                                <div class="headline">ALLMÄNT</div>
                                <div class="ingress">
                                    Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                                </div>
                            </div>
                            <div key="1">
                                <div class="headline">ÄGA</div>
                                <div class="ingress">
                                    Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                                </div>
                            </div>
                            <div key="2">
                                <div class="headline">HYRA</div>
                                <div class="ingress">
                                    Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                                </div>
                            </div>
                            <div key="3">
                                <div class="headline">RESULTAT</div>
                                <div class="ingress">
                                    Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                                </div>
                            </div>
                            <div key="/">
                                <div class="col1span8 headline">FALLBACK</div>
                            </div>
	                    </iron-pages>
                </div>


                <div id="commonsection" class="col1span12">
                    <whcg-section-wide-chartlong>
                        <span slot="title">TABELL</span>
                        <span slot="text">Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                        </span>
                        <whcg-lit-grid slot="chart" items="{{maintableitems}}">
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column width="60px" flex-grow="0">
                                <template class="header">#</template>
                                <template>[[index]]</template>
                            </vaadin-grid-column>
                            <vaadin-grid-column width="140px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="raw_real_estate_company_name">Mäklare</vaadin-grid-sorter></template>
                                <template>[[item.raw_real_estate_company_name]]</template>
                            </vaadin-grid-column>
                            <vaadin-grid-column width="60px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="municipality">Kommun</vaadin-grid-sorter></template>
                                <template>[[item.municipality]]</template>
                            </vaadin-grid-column>

                            <vaadin-grid-column width="10px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="build_year">Byggår</vaadin-grid-sorter></template>
                                <template>[[item.build_year]]</template>
                            </vaadin-grid-column>

                            <vaadin-grid-column width="60px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="formatted_address">Adress</vaadin-grid-sorter></template>
                                <template>[[item.formatted_address]]</template>
                            </vaadin-grid-column>
                            <vaadin-grid-column width="60px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="type_of_housing">Hustyp</vaadin-grid-sorter></template>
                                <template>[[item.type_of_housing]]</template>
                            </vaadin-grid-column>
                            <vaadin-grid-column width="60px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="contract_date">Kontraktsdatum</vaadin-grid-sorter></template>
                                <template>[[item.contract_date]]</template>
                            </vaadin-grid-column>
                            <vaadin-grid-column width="60px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="contract_price">Köpeskilling</vaadin-grid-sorter></template>
                                <template>[[item.contract_price]]</template>
                            </vaadin-grid-column>                             
                        </whcg-lit-grid>
                    </whcg-section-wide-chartlong>






                    <whcg-section-textlong-input-chart>
                        <span slot="text">Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                        </span>
                        <span slot="title">INITIAL ETABLERINGSKOSTNAD</span>
                        <whcg-lit-grid slot="input1" items="{{brokertableitems}}">
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column width="140px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="key">Mäklare</vaadin-grid-sorter></template>
                                <template>[[item.key]]</template>
                            </vaadin-grid-column>                    
                        </whcg-lit-grid>
                       <whcg-lit-grid slot="input2" items="{{municipalitytableitems}}">
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column width="140px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="key">Kommun</vaadin-grid-sorter></template>
                                <template>[[item.key]]</template>
                            </vaadin-grid-column>                    
                        </whcg-lit-grid>
                        <whcg-lit-grid slot="input3" items="{{brokertableitems}}">
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column width="140px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="key">Mäklare</vaadin-grid-sorter></template>
                                <template>[[item.key]]</template>
                            </vaadin-grid-column>                    
                        </whcg-lit-grid>
                       <whcg-lit-grid slot="input4" items="{{municipalitytableitems}}">
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column width="140px" flex-grow="1" flex-shrink="1" resizable>
                                <template class="header"><vaadin-grid-sorter path="key">Kommun</vaadin-grid-sorter></template>
                                <template>[[item.key]]</template>
                            </vaadin-grid-column>                    
                        </whcg-lit-grid>
                    </whcg-section-textlong-input-chart>


                    <!--
                    <whcg-section-text-input>
                        <span slot="title">KALKYLLÄNGD</span>
                        <span slot="text">Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                        </span>
                        <vaadin-grid slot="input" id="namegrid" aria-label="Basic Binding Example" items="{{nametabledata}}" on-change="handleNameGridChangeFind" >
                            <vaadin-grid-selection-column auto-select></vaadin-grid-selection-column>
                            <vaadin-grid-column>
                            <template class="header"><vaadin-grid-sorter path="key">Name</vaadin-grid-sorter></template>
                            <template>[[item.key]]</template>
                            <template class="footer">Name</template>
                            </vaadin-grid-column>
                        </vaadin-grid>
                    </whcg-section-text-input>
                    -->

                    <whcg-section-input-text>
                        <span slot="title">INFLATION</span>
                        <whcg-number-field-box slot="input" name="Inflation" mode="none">
                            <whcg-select label="Inflation" suffix="%" value="0.02" placeholder="...antal procent" valueoutput="{{inflation}}" jsoninput='[{"value": 0.01, "caption": "1"}, {"value": 0.02, "caption": "2"}, {"value": 0.03, "caption": "2"}, {"value": 0.04, "caption": "4"}, {"value": 0.05, "caption": "5"}, {"value": 0.06, "caption": "6"}, {"value": 0.07, "caption": "7"}, {"value": 0.08, "caption": "8"}, {"value": 0.09, "caption": "9"}, {"value": 0.10, "caption": "10"}]'></whcg-select>
                        </whcg-number-field-box>
                        <span slot="text">Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                        </span>
                    </whcg-section-input-text>

                    <whcg-section-text-input>
                        <span slot="title">KALKYLRÄNTA</span>
                        <span slot="text">Pellentesque sit amet nisl odio. Duis erat libero, placerat vitae mi at, bibendum porta nisi. Proin fermentum mi et nibh sollicitudin, in interdum mauris molestie. Aliquam fermentum dolor pulvinar tempus blandit. Cras aliquam lectus ut dolor ornare aliquam. Curabitur lobortis ut nibh in sollicitudin. In viverra facilisis magna, a tempus lorem dictum at. Ut porta vehicula lacus, nec mollis libero rutrum id. Aliquam quis tristique risus.
                        </span>
                        <whcg-number-field-box slot="input" name="Kalkylränta" mode="none">
                            <whcg-select label="Kalkylränta" suffix="%" value="0.07" placeholder="...antal procent" valueoutput="{{discountrate}}" jsoninput='[{"value": 0.01, "caption": "1"}, {"value": 0.02, "caption": "2"}, {"value": 0.03, "caption": "2"}, {"value": 0.04, "caption": "4"}, {"value": 0.05, "caption": "5"}, {"value": 0.06, "caption": "6"}, {"value": 0.07, "caption": "7"}, {"value": 0.08, "caption": "8"}, {"value": 0.09, "caption": "9"}, {"value": 0.10, "caption": "10"}]'></whcg-select>
                        </whcg-number-field-box>
                    </whcg-section-text-input>
                </div>


        </div>
        <div class="right"></div>
    </div>
    `;
  }


  connectedCallback() {
      super.connectedCallback();
      let that = this;

      this.db.find({
          //selector: { raw_real_estate_company_name: { $gte: "a" } },
          //selector: { municipality: { $eq: "NACKA" } },
          // selector: {
          //   municipality: {$or: [
          //     { $eq: 'STOCKHOLM' },
          //     { $eq: 'NACKA' }
          //   ]}
          // },
          selector: { 
            raw_real_estate_company_name: { $gte: "a" },
            municipality: { $gte: null },
            build_year: { $gte: "1955" }
          },
          limit: 1000
      }).then(function (result) {
          console.log(result);
          console.log(result.docs);
          that.maintableitems = result.docs;
      }).catch(function (err) {
          console.log(err);
      });




      this.db.query('my_index8/by_name', {
          startkey: 'A',
          reduce: true,
          group: true,
      }).then(function (res) {

          let rows = res.rows;

          that.brokertableitems = rows;
      }).catch(function (err) {
          console.log(err);
      });


      this.db.query('my_index8/by_municipality', {
          startkey: 'A',
          reduce: true,
          group: true,
      }).then(function (res) {

          let rows = res.rows;

          that.municipalitytableitems = rows;
      }).catch(function (err) {
          console.log(err);
      });

      this.db.query('my_index8/by_build_year', {
          startkey: '1',
          reduce: true,
          group: true,
      }).then(function (res) {

          let rows = res.rows;

          that.municipalitytableitems = rows;
      }).catch(function (err) {
          console.log(err);
      });
  }
}

window.customElements.define("whcg-app", WhcgApp);

export { WhcgApp };
